### ISSUE TITLE
SSHSessionProcessProtocol does not implement getHost/getPeer.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2453]
|<img alt="jrydberg's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| jrydberg reported|
|-|-|
|Trac ID|trac#2453|
|Type|defect|
|Created|2007-02-14 15:16:54Z|
|Branch|https://github.com/twisted/twisted/tree/session-transport-addresses-2453|

session.SSHSessionProcessProtocol does not implement getHost/getPeer methods
of the transport interface.

Attachments:

* [twisted-2453.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/caa/caaa9552af2f6811278fe10f8f42beece2a72fdd/01b0dafceca8d534feeff10daf0e431f03ac10c8.diff) (4248 bytes) - added by salgado on 2011-03-15 14:33:27Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2453 2453
type__defect defect
reporter__jrydberg jrydberg
priority__normal normal
milestone__ 
branch__branches_session_transport_addresses_2453 branches/session-transport-addresses-2453
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__soc2007_easy soc2007 easy
time__1171466214000000 1171466214000000
changetime__1300217678000000 1300217678000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|z3p commented|
|-|-|

Tagging soc2007

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

These should return some sort of `IAddress`.  An easy solution would be to just return the host/peer that the corresponding `getHost` or `getPeer` method on the wrapped `session.conn.transport` would return (of course it's somewhat gross to reach that far out of the `SSHSessionProcessProtocol`... but it's not exactly surprising in Conch).

**Comment by twisted-trac** [comment]
|<img alt="salgado's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|salgado commented|
|-|-|

I'm probably missing something, but it confuses me that SSHSessionProcessProtocol should implement methods of the transport interface when it doesn't seem to provide ITransport?

**Comment by twisted-trac** [comment]
|<img alt="salgado's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|salgado set owner to salgado<br>salgado set status to `assigned`|
|-|-|

I discussed this with exarkun and it seems that SSHSessionProcessProtocol should be marked as implementing ITransport, so I may do that as well.

**Comment by twisted-trac** [comment]
|<img alt="salgado's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|salgado commented|
|-|-|

This is my first attempt at adding getHost/getPeer to SSHSessionProcessProtocol. I couldn't resist adding/removing some blank lines to make the surroundings of what I changed follow the current policy (2 lines between methods and 3 between top-level classes is what I was told).

**Comment by twisted-trac** [comment]
|<img alt="salgado's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|salgado removed owner<br>salgado set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [31141]) Branching to 'session-transport-addresses-2453'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [31142]) Apply patch from the ticket

refs [#2453](https://github.com/twisted/twisted/issues/2453)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

Great, thanks!  The tests and implementation look good, except for `assertEqual` which I'll change to `assertEquals` before merging.  [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/session-transport-addresses-2453) also look good (at least, as good as I expect them to look at the moment).  So I'll go ahead and apply this.  Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set status to `closed`|
|-|-|

(In [31146]) Merge session-transport-addresses-2453

Author: salgado
Reviewer: exarkun
Fixes: [#2453](https://github.com/twisted/twisted/issues/2453)

Add `getHost` and `getPeer` methods to `SSHSessionProcessProtocol`, used as the transport
for subsystem protocols.
